src/platform.rs

```diff
@@ -43,9 +43,17 @@ impl Platform {
                 format!("{}/{}", left, right)
             }
             Platform::Win => {
-                let left = left.trim_end_matches('\\');
-                let right = right.trim_start_matches('\\');
-                format!("{}\\{}", left, right)
+                // We need to support both Linux-style paths and Windows-style paths.
+                // That's because Platform::Win is used for all PDB files, even for
+                // PDB files for build that were (cross-)compiled on a Linux machine;
+                // those contain Linux paths.
+                let left = left.trim_end_matches(&['/', '\\']);
+                let right = right.trim_start_matches(&['/', '\\']);
+                if left.starts_with('/') {
```

Comment:
This will work only for absolute paths, is this something we always guarantee [here](https://github.com/mozilla/dump_syms/blob/e2640b47927baa8e616d1f1f31d78cdb428d5301/src/source.rs#L68-L70)? Or at least do we always guarantee it for Linux?

Reply:
It depends on what's in the PDB, but the xul.pdb I looked at only contained absolute paths.

I can add a comment saying that this is a bit of a hack and only works in the absolute path case.

Reply:
I've added the comment.